charts/node/templates/statefulset.yaml

```diff
@@ -79,183 +79,150 @@ spec:
           args:
             - -c
             - |
-              set -eu -o pipefail {{ if .Values.initContainer.debug }}-x{{ end }}
-              if [ -d "/relaychain-data/chains/${RELAY_CHAIN_PATH}/{{ $databasePath }}" ]; then
-                echo "Database directory already exists, skipping relay-chain snapshot download"
+              set -eu -o pipefail {{ if .Values.initContainers.downloadChainSnapshot.debug }}-x{{ end }}
+              if [ -d "/chain-data/chains/${RELAY_CHAIN_PATH}/{{ $databasePath }}" ]; then
+                echo "Database directory already exists, skipping chain snapshot download"
               else
-                echo "Downloading relay-chain snapshot"
-                RELAY_SNAPSHOT_URL="{{ .Values.node.collatorRelayChain.chainData.snapshotUrl }}"
-                mkdir -p /relaychain-data/chains/${RELAY_CHAIN_PATH}/{{ $databasePath }}
-                if [ "${RELAY_SNAPSHOT_FORMAT}" == "lz4" ]; then
-                  wget -O - ${RELAY_SNAPSHOT_URL} | lz4 -c -d - | tar -x -C /relaychain-data/chains/${RELAY_CHAIN_PATH}/
-                else
-                  wget -O - ${RELAY_SNAPSHOT_URL} | tar -x -C /relaychain-data/chains/${RELAY_CHAIN_PATH}/{{ $databasePath }}/
+                PARALLEL_TRANFERS="$(($(nproc --all) * 5 < 50 ? $(nproc --all) * 5 : 50))" # MIN(vCPU_count * 5, 50)
+                echo "Downloading chain snapshot"
+                SNAPSHOT_URL="{{ .Values.node.collatorRelayChain.chainData.chainSnapshot.url }}"
+                mkdir -p /chain-data/chains/${RELAY_CHAIN_PATH}/{{ $databasePath }}/
+
+                if [ "${METHOD}" == "http-single-tar-lz4" ]; then
+                  apk add lz4 --no-cache
+                  rclone copyurl {{ .Values.initContainers.downloadChainSnapshot.cmdArgs }} --stdout --error-on-no-transfer ${SNAPSHOT_URL} | lz4 -c -d - | tar -x -C /chain-data/chains/${RELAY_CHAIN_PATH}/
+                  chown -R {{ .Values.podSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }} /chain-data/chains/${RELAY_CHAIN_PATH}/
+
+                elif [ "${METHOD}" == "http-single-tar" ]; then
+                  rclone copyurl {{ .Values.initContainers.downloadChainSnapshot.cmdArgs }} --stdout --error-on-no-transfer ${SNAPSHOT_URL} | tar -x -C /chain-data/chains/${RELAY_CHAIN_PATH}/
+
+                elif [ "${METHOD}" == "gcs" ]; then
+                  LATEST=$(rclone cat {{ .Values.initContainers.downloadChainSnapshot.cmdArgs }} --quiet :gcs:${SNAPSHOT_URL}/latest_version.meta.txt)
+                  if [ -z "$LATEST" ]; then
+                    echo "Failed to retrieve latest_version.meta.txt file. Will download everything from ${SNAPSHOT_URL} instead"
+                  fi
+                  rclone sync {{ .Values.initContainers.downloadChainSnapshot.cmdArgs }} --fast-list --transfers $PARALLEL_TRANFERS --progress --error-on-no-transfer :gcs:${SNAPSHOT_URL}/${LATEST} /chain-data/chains/${RELAY_CHAIN_PATH}/{{ $databasePath }}/
```

Comment:
Does it work with public buckets without the [option](https://rclone.org/googlecloudstorage/#gcs-bucket-policy-only). I mean buckets that use Uniform bucket-level access.